axelor-sale/src/main/java/com/axelor/apps/sale/service/SaleOrderGeneratorServiceImpl.java

```diff
@@ -116,4 +127,25 @@ protected void checkContact(Partner clientPartner, Partner contactPartner)
           I18n.get(SaleExceptionMessage.CONTACT_PROVIDED_DOES_NOT_RESPECT_DOMAIN_RESTRICTIONS));
     }
   }
+
+  @Transactional(rollbackOn = {Exception.class})
+  @Override
+  public SaleOrder fetchAndAddSaleOrderLines(
+      List<SaleOrderLinePostRequest> saleOrderLinePostRequests, SaleOrder saleOrder)
+      throws AxelorException {
+    List<SaleOrderLine> saleOrderLineList = new ArrayList<>();
```

Comment:
Check directly here if saleOrderLinePostRequests is empty, if yes, return directly the sale order.